### PR TITLE
Updated SSO

### PR DESCRIPTION
**Pull request by jerae-duffin**
Updated SSO docs

### CODE REVIEW

File: _data/toc.yaml [hunk 1]
```diff
@@ -1386,6 +1386,12 @@ manuals:
             title: Recover your Docker Hub account
           - path: /docker-hub/2fa/new-recovery-code/
             title: Generate a new recovery code
+      - sectiontitle: Single Sign-on
+        section:
+          -  path: /single-sign-on/
+             title: Single-Sign-on
```
Comment (usha-mandya) [inline _data/toc.yaml:1392]:
```suggestion
             title: Single Sign-on
```

File: Single-Sign-on/faqs.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Yes, all users in your organization must upgrade to Docker Desktop version 4.4.0 or higher. Users on older versions of Docker Desktop will not be able to sign in after enforcing SSO.
+
+### Q: Does SAML authentication require additional attributes?
+
+You must provide an email address as an attribute to authenticate via SAML. The ‘Name’ attribute is currently optional.
+
+### Q: Does the application recognize the NameID/Unique Identifier in the SAMLResponse Subject? If so please provide the format you require.
+
+TBD
```
Comment (usha-mandya) [inline Single-Sign-on/faqs.md:83]: Could you remove this if we don't have any info yet?

**Comment by usha-mandya** [comment]
Looks like there are broken links. Could you check the details in the build checks and fix those?

File: _data/toc.yaml [hunk 3]
```diff
@@ -1386,6 +1386,12 @@ manuals:
...
+          -  path: /single-sign-on/faqs
```
Comment (usha-mandya) [inline _data/toc.yaml:1393]: As the title of this section is also Single Sign-on, can we call this Configure SSO?

Comment (usha-mandya) [reply _data/toc.yaml:1393]: image

Comment (jerae-duffin) [reply _data/toc.yaml:1393]: Good catch! Yes, updating it now @usha-mandya

File: Single-Sign-on/faqs.md (hunk 2 above)
Comment (jerae-duffin) [reply Single-Sign-on/faqs.md:83]: Deleting it for now @usha-mandya

File: Single-Sign-on/faqs.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
+---
+description: Single Sign-on FAQs
+keywords: Docker, Docker Hub, SSO FAQs, single sign-on
+title: SSO FAQs
+---
```
Comment (usha-mandya) [inline Single-Sign-on/faqs.md:5]: Could you add the following to the metadata so the FAQ doesn't list all the questions in the right navigation (mini toc on the right)

```
toc_max: 2
```

Comment (jerae-duffin) [reply Single-Sign-on/faqs.md:5]: Updated

File: Single-Sign-on/faqs.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+* **SAML 2.0**: Entity ID, ACS URL, Single Logout URL and Certificate Download URL
+* **Azure AD**: Client ID, Client Secret, AD Domain
+
+### Q: Is DNS verification required to enable SSO?
+
+Yes. You must verify a domain before using it with an SSO connection.
+
+### Q: Does Docker SSO support authenticating through the command line?
+
+Yes. When SSO is enabled, you can access the Docker CLI through Personal Access Tokens (PATs).  Each user must create a PAT to access the CLI. To learn how to create a PAT, see [Managing access tokens](/docker-hub/access-tokens/).
```
Comment (izzychu) [inline Single-Sign-on/faqs.md:66]: Can we add: Before we transition to PATs. CLI users will continue to be able to log in using their personal credentials until early next year to mitigate risk of interrupting CI/CD pipelines.

Comment (jerae-duffin) [reply Single-Sign-on/faqs.md:66]: Updated @izzychu

File: Single-Sign-on/faqs.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+### Q: Do I need to manually add users to my organization?
+
+No, you don’t need to manually add users to your organization in Docker Hub. You just need to make sure an account for your users exists in your IdP and then invite them to your organization using the **Invite Member** option in Docker Hub.
+
+When a user signs into Docker for the first time using their domain email address, they will be automatically added to the organization after a successful authentication.
+
+### Q: Can users in my organization use different email addresses to authenticate via SSO?
+
+During the SSO setup, you’ll have to specify the company email domains that are allowed to authenticate. All users in your organization must authenticate using the email domain specified during SSO setup.
```
[the remark this answers is not in the document]
Comment (jerae-duffin) [reply Single-Sign-on/faqs.md:100]: Updated

File: Single-Sign-on/index.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+To enable SSO in Docker Hub, you need the following:
+
+* **SAML 2.0**: Entity ID, ACS URL, Single Logout URL and Certificate Download URL
+* **Azure AD**: Client ID (a unique identifier for your registered AD application), Client Secret (a string used to gain access to your registered Azure AD application), and AD Domain details
+
+We currently support enabling SSO on a single organization. If you have any users in your organization with a different domain (including social domains), they will be added to the organization as guests.
+
+## SSO prerequisites
+
+* You must first notify your company about the new SSO login procedures
```
Comment (izzychu) [inline Single-Sign-on/index.md:24]: We can insert the note about users may wish to maintain separate account for personal and company use?

File: Single-Sign-on/faqs.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+For step by step instructions on how to configure Docker SSO, see [Single Sign-on docs](/single-sign-on/#index.md).
+
+### Q: Is it possible to use more than one IdP with Docker SSO?
+
+No. You can only configure Docker SSO to work with a single IdP. A domain can only be associated with a single IdP.Docker currently supports Azure AD and identity providers that support SAML 2.0
+
+### Q: Is it possible to change my identity provider after configuring SSO?
+
+Yes. You must delete your existing IdP configuration in Docker Hub and follow the instructions in the [Single Sign-on docs](/single-sign-on/#index.md) to configure SSO using your IdP.
```
Comment (usha-mandya) [inline Single-Sign-on/faqs.md:36]:
```suggestion
Yes. You must delete your existing IdP configuration in Docker Hub and follow the instructions in the [Single Sign-on docs](/single-sign-on/index.md) to configure SSO using your IdP.
```

File: Single-Sign-on/faqs.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+You can enable SSO on organizations that are part of the Docker Business subscription.
+
+### Q: We currently have a Docker Team subscription. How do we enable SSO?
+
+Docker SSO is available with a Docker Business subscription. To enable SSO, you must first upgrade your subscription to a Docker Business subscription. To learn how to upgrade your existing account, see [Upgrade your subscription](/subscription/upgrade/){:target="blank" rel="noopener" class=""}.
+
+### Q: Where can I find detailed instructions on how to configure Docker SSO?
+
+For step by step instructions on how to configure Docker SSO, see [Single Sign-on docs](/single-sign-on/#index.md).
```
Comment (usha-mandya) [inline Single-Sign-on/faqs.md:28]:
```suggestion
For step by step instructions on how to configure Docker SSO, see [Single Sign-on docs](/single-sign-on/index.md).
```

File: Single-Sign-on/faqs.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+### Q: Is it possible to convert existing users from non-SSO to SSO accounts?
+
+Yes, you can convert existing users to an SSO account. To convert users from a non-SSO account:
+
+* Ensure your users have a company domain email address and they have an account in your IdP
+* Verify that all users have Docker Desktop version 4.4.0 or higher installed on their machines
+* Each user has created a PAT to replace their passwords to allow them to log in through Docker CLI
+* Confirm that all CI/CD pipelines automation systems have replaced their passwords with PATs.
+
+For detailed prerequisites and for instruction on how to enable SSO, see [Single Sign-on](/single-sign-on/#index.md).
```
Comment (usha-mandya) [inline Single-Sign-on/faqs.md:131]:
```suggestion
For detailed prerequisites and for instruction on how to enable SSO, see [Single Sign-on](/single-sign-on/#index.md).
```
```suggestion
For detailed prerequisites and for instructions on how to enable SSO, see [Configure Single Sign-on](index.md).
```

Comment (usha-mandya) [inline Single-Sign-on/faqs.md:131]:
```suggestion
For detailed prerequisites and for instructions on how to enable SSO, see [Configure Single Sign-on](#index.md).
```

**Review by izzychu — APPROVED** [review]

**Comment by usha-mandya** [comment]
@jerae-duffin The ToC still appears broken. Could be due to the case sensitivity. Approving now so it's not a blocker for release. Let's get this fixed in a follow-up PR if it still appears broken

**Review by usha-mandya — APPROVED** [review]

**Pull request merged**